microsetta_public_api/config.py

```diff
@@ -1,6 +1,226 @@
 import os
 import json
-from microsetta_public_api.exceptions import ConfigurationError
+from typing import Union
+import jsonschema
+
+from microsetta_public_api.backend import constructors
+
+
+def component_schema(properties=None,
```

Comment:
Pydantic looks super cool! And has some [nifty examples of integration with SQLAlchemy](https://pydantic-docs.helpmanual.io/usage/models/#orm-mode-aka-arbitrary-class-instances).

I decided to go with JSON-schema here since everyone already has to be familiar with it to edit those api `yaml` files anyways.